milatools/utils/vscode_utils.py

```diff
@@ -60,130 +57,128 @@ def get_code_command() -> str:
+    if not destinations:
+        logger.info("No destinations to sync extensions to!")
+        return {}
 
-    This spawns a thread for each cluster in `dest` and displays a parallel progress bar
-    for the syncing of vscode extensions to each cluster.
-    """
-    if isinstance(source, LocalV1):
-        source_hostname = "localhost"
-        source_extensions = get_local_vscode_extensions()
-    elif isinstance(source, RemoteV2):
-        source_hostname = source.hostname
-        code_server_executable = find_code_server_executable(
-            source, remote_vscode_server_dir="~/.vscode-server"
-        )
-        if not code_server_executable:
-            raise RuntimeError(
-                f"The vscode-server executable was not found on {source.hostname}."
-            )
-        source_extensions = get_remote_vscode_extensions(source, code_server_executable)
-    else:
-        assert isinstance(source, str)
-        source_hostname = source
-        source = RemoteV2(source)
+    source_extensions = await _get_vscode_extensions(source)
 
     task_hostnames: list[str] = []
-    task_fns: list[TaskFn[ProgressDict]] = []
+    tasks: list[AsyncTaskFn[list[str]]] = []
     task_descriptions: list[str] = []
 
-    for dest_remote in dest_clusters:
-        dest_hostname: str
-
-        if dest_remote == "localhost":
-            dest_hostname = dest_remote  # type: ignore
-            dest_remote = LocalV1()  # pickleable
-        elif isinstance(dest_remote, LocalV1):
-            dest_hostname = "localhost"
-            dest_remote = dest_remote  # again, pickleable
-        elif isinstance(dest_remote, RemoteV2):
-            dest_hostname = dest_remote.hostname
-            dest_remote = dest_remote  # pickleable
-        elif isinstance(dest_remote, RemoteV1):
-            # We unfortunately can't pass this kind of object to another process or
-            # thread because it uses `fabric.Connection` which don't appear to be
-            # pickleable. This means we will have to re-connect in the subprocess.
-            dest_hostname = dest_remote.hostname
-            dest_remote = None
-        else:
-            assert isinstance(dest_remote, str)
-            # The dest_remote is a hostname. Try to connect to it with a reusable SSH
-            # control socket so we can get the 2FA prompts out of the way in advance.
-            # NOTE: We could fallback to using the `Remote` class with paramiko inside
-            # the subprocess if this doesn't work, but it would suck because it messes
-            # up the UI, and you need to press 1 in the terminal to get the 2FA prompt,
-            # which screws up the progress bars.
-            dest_hostname = dest_remote
-            dest_remote = RemoteV2(hostname=dest_hostname)
-
+    # Connect to the remotes in parallel.
+    dest_runners_and_hostnames = await asyncio.gather(
+        *(_get_runner_and_hostname(dest) for dest in destinations)
+    )
```

Comment:
It does not seam necessary to have this bit of coda and the `_get_runner_and_hostname` util as `_install_vscode_extensions_task_function` will connect in parallel anyway?

Reply:
Fixed in 94f690ddd6445b89f8f43922a46f61af4a443a9d